util/java/libtiled-java/src/main/java/tiled/core/Map.java

```diff
@@ -304,8 +402,19 @@ public String getFilename() {
      * Returns a vector with the currently loaded tileSets.
      *
      * @return Vector
+     * @deprecated Use getMapTileSets instead.
```

Comment:
I wasn't too sure what level of API compatability you were after, If there's no compatibility guarantees, i've removed the deprecated methods and just changed the return type as suggested, I've also removed the commits that added the Ant Scripts and changes to junit.

As for the comment for adding support for multi linked images in tilesets, I have added a new method in TMXMapWriter, "writeImage" and a new property in Tile (with getter and setter), "source" that allows to write individual tiles to a tilemap as links to external files. This creates an identical format to using Type "Collection of Images" when creating a new Tileset in the Tiled application v0.17. e.g:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<tileset name="TileSet" tilewidth="32" tileheight="32" tilecount="2123" columns="0">
 <tile id="0">
  <image width="32" height="32" source="tiles/0.png"/>
 </tile>
</tileset>
```